script.js

```diff
@@ -0,0 +1,164 @@
+
+class RenderManager {
+    list = new ListManager();
+    submitButton = document.getElementById("addMission");
+    clearAllButton = document.getElementById("clear-all");
+    unFinishedConteiner = document.getElementById("unfinishedMissions")
+    finishedConteiner = document.getElementById("finishedMissions")
+
+    constructor() {
+        this.submitButton.addEventListener("click", this.handleNewItem.bind(this));
+        this.clearAllButton.addEventListener("click", () => {this.handleClearAll();});
+    }
+
+    render() {
+        this.finishedConteiner.innerHTML = "";
+        this.unFinishedConteiner.innerHTML = "";
+        for(const miss of this.list.missions){
+            const element = this.createMissionElement(miss);
+            if(miss.is_finished){
+                this.addMissionToFinished(element);
+            }else{
+                this.addMissionToUnfinished(element);
+            }
+        }
+    }
+ 
+    handleClearAll() {
+        const confirmation = confirm("Are you sure about it?");
+        if(confirmation){
+            this.clearAllLists();
+        }
+    }
+
+    clearAllLists() {
+        this.finishedConteiner.innerHTML = "";
+        this.unFinishedConteiner.innerHTML = "";
+        this.list = [];
+    }
+
+    validateDate(inputDate) {
+        const year = Number(inputDate.split('-')[0]);
+
+        if(year < 2000 || year > 2030){
+            alert("year must be in range [2000, 2030]");
+            return false;
+        }
+        return true;
+    }
+
+    formatDateToDDMMYYYY(inputDate){
+        const [year, month, day] = inputDate.split('-');
+        return `${day}/${month}/${year}`;
+    }
+
+
+    handleNewItem() {
+        const name = document.getElementById("missionName").value.trim();
+        const date = document.getElementById("missionDate").value;
+
+        if (!this.validateDate(date)){return;}
```

Comment:
don't you want to alert on this as well?

Reply:
i throw alert in the function
its not readable so i change it now.